src/pubsub/ua_pubsub_manager.c

```diff
@@ -48,6 +48,23 @@ UA_Server_addPubSubConnection(UA_Server *server,
         return retval;
     }
 
+    if(UA_Variant_isEmpty(&tmpConnectionConfig->publisherId)){
+        UA_Byte publisherId = 0;
+        UA_Variant_setScalarCopy(&tmpConnectionConfig->publisherId, &publisherId, &UA_TYPES[UA_TYPES_BYTE]);
+    }
```

Comment:
I couldn't find anything regarding this in the standard, since there is a possibility to neglect sending a publisherId, it could be possible to default to disableing the corresponding header entries. A lot of tests didn't bother to specify an Id and since the initializied value was numeric with 0 it worked in their cases. I did this to not break the tests otherwise i would have been rejecting an empty variant.